src/BuildingRegistry/Building/Building_BuildingUnit.cs

```diff
@@ -220,34 +195,35 @@ private void AddCommonBuildingUnit(IAddCommonBuildingUnit addCommonBuildingUnit)
             addCommonBuildingUnit.AddForBuilding(BuildingPersistentLocalId, commonBuildingUnitPersistentLocalId);
         }
 
-        private void UpdateStatusCommonBuildingUnit()
+        private void CorrectCommonBuildingUnitToPlannedOrRealized()
         {
-            var commonBuildingUnit = _buildingUnits.CommonBuildingUnit();
+            var commonBuildingUnit = _buildingUnits.CommonBuildingUnit(excludeRemoved: false);
 
-            void CheckToUpdateCommonBuildingUnit()
+            if (commonBuildingUnit.IsRemoved)
```

Comment:
@ArneD would you move this if branch to `EnsureCommonBuildingUnit()` or leave it here?

Reply:
Yes, that would make sense.

Reply:
Changed.